.coveragerc

```diff
@@ -0,0 +1,19 @@
+[run]
+branch = True
+
+[report]
+# Regexes for lines to exclude from consideration
+exclude_lines =
+    # Don't complain if non-runnable code isn't run:
+    if 0:
```

Comment:
It is a good idea to add:
- `raise NotImplementedError`
- `r'^ +\.\.\.$',`
- `r'^\s*if (False|TYPE_CHECKING):',`

See https://github.com/asottile/covdefaults for more useful ideas.

Reply:
This file is currently the coverage default + IDLE additions, designed to be unlikely to affect other files (but not guaranteed, I admit, which is why I labelled them) + checking branches both ways.  Anything added should be something that should always be excluded from a whole-suite coverage test.

* NotImplementedError: It seems to me that test of abstract methods might use `assertRaises(NotImplentedError, ...)`.
* Re appears to match ` ...` line.  How might this be used in code and why would this never by executed by any test?
* False == 0 so I will add that.  Arguably should be in coverage default though `if 0:` is traditional and easier to add for temporary disablement.
* Please explain `TYPE_CHECKING` and its use.  (Being all caps it should be a constant, always true or always false, and thus not suitable as a conditional.)

Reply:
I think the current PR is a good starting point; IMO we should land it pretty much as it is.

We can discuss and cherry-pic regexes from Anthony's config in a follow up PR. For example, I'm not sure about the NotImplemented case for the reasons Terry provided.